# Parte Conceptual - Prueba Técnica Totto
### Autor: Daniel Fernando Manosalva Cáceres
### Fecha: Agosto 2025

Este cuaderno contiene el desarrollo de la sección conceptual de la prueba técnica para el cargo de Analista Comercial.

## 1. ¿Pueden A y B ser mutuamente excluyentes?
- P(A) = 1/2, P(B^C) = 1/4  P(B) = 3/4
- Para que sean mutuamente excluyentes: P(A ∩ B) = 0, sin embargo, la probabilidad total no puede superar 1.
- P(A∩B)=P(A)+P(B)−P(A∪B)≤P(A)+P(B)= 1/2 + 3/4 =5/4

**Respuesta:** No pueden ser mutuamente excluyentes porque la suma de sus probabilidades excede 1.

## 2. Control de Calidad - Barras de Chocolate
**Datos:**
- Tamaño muestra: 15
- σ conocida: 2
- μ objetivo: 100
- α: 0.05

In [7]:
import numpy as np
from scipy.stats import norm

barras = np.array([94.40, 97.64, 98.48, 97.67, 100.11,
                  95.29, 99.80, 98.80, 100.53, 99.41,
                  97.64, 101.11, 93.43, 96.99, 97.92])

n = len(barras)
sigma = 2
mu = 100
x_bar = np.mean(barras)
z_2_colas = (x_bar - mu) / (sigma / np.sqrt(n))
p_2_colas = 2 * norm.sf(abs(z_2_colas))

z_izquierda = z_2_colas
p_izquierda = norm.cdf(z_izquierda)

print(f"Media muestral: {x_bar:.2f}")
print(f"Z (dos colas): {z_2_colas:.3f}, p-valor: {p_2_colas:.6f}")
print(f"Z (una cola): {z_izquierda:.3f}, p-valor: {p_izquierda:.6f}")

Media muestral: 97.95
Z (dos colas): -3.974, p-valor: 0.000071
Z (una cola): -3.974, p-valor: 0.000035


#### a) ¿Cuáles son las hipótesis respecto al peso esperado μ para una prueba de dos colas?
Z de dos colas
- Estadístico Z: -3.97
- p-valor: 0.000071
Rechazamos 𝐻0. Hay evidencia significativa de que el peso promedio no es igual a 100g.

#### b) ¿Qué prueba debe utilizarse para contrastar estas hipótesis?
Se conoce 𝜎=2 (desviación estándar poblacional) y n = 15.
Usamos la prueba Z de una muestra (no T porque σ es conocida).

#### c) Realice la prueba que sugirió en (b). Use α = 0,05.
Comparar con el valor crítico Z (α/2 = 0.025) =  ±1.96

#### d) El productor quiere demostrar que el peso esperado es menor que 100 g. ¿Cuáles son las hipótesis apropiadas para usar?
Es una prueba de cola izquierda porque el interés está en detectar una disminución respecto al valor de referencia.
Z de una cola (izquierda)
- Estadístico Z: -3.97
- p-valor: 0.000035
Rechazamos 𝐻0. Hay evidencia para afirmar que el peso promedio es menor a 100g.

#### e) Realice la prueba para la hip´otesis en (d). Nuevamente use α = 0,05.
Como el estadístico Z = -3.974 < -1.645 y el p-valor < 0.05, se rechaza la hipótesis nula.


### Interpretación:
- Z = -3.97, p < 0.05
- Rechazamos H0 en ambos casos (dos colas y una cola)
- **Conclusión:** Existe evidencia estadística suficiente para afirmar que el peso promedio es menor que 100 gramos. 

## 3. Regresión - Horas de Sueño vs Internet

a) Estimar el modelo de regresión lineal

In [5]:
from sklearn.linear_model import LinearRegression
from sklearn.metrics import r2_score

internet = np.array([0.3, 2.2, 0.5, 0.7, 1.0, 1.8, 3.0, 0.2, 2.3])
sueno = np.array([5.8, 4.4, 6.5, 5.8, 5.6, 5.0, 4.8, 6.0, 6.1])

# Regresión continua
X_cont = internet.reshape(-1, 1)
y = sueno
reg_cont = LinearRegression().fit(X_cont, y)
y_pred_cont = reg_cont.predict(X_cont)
r2_cont = r2_score(y, y_pred_cont)

print("Modelo continuo:")
print(f"Beta_0 (intercepto): {reg_cont.intercept_:.3f}")
print(f"Beta_1 (pendiente): {reg_cont.coef_[0]:.3f}")
print(f"R²: {r2_cont:.3f}")

Modelo continuo:
Beta_0 (intercepto): 6.155
Beta_1 (pendiente): -0.450
R²: 0.448


b) Calcule R2 para evaluar el grado de ajuste del modelo.

R² ≈ 0.448

**Interpretación:**

β1 = −0.45: por cada hora adicional en internet, el sueño disminuye 0.45 horas (efecto negativo).

𝛽0 = 6.155: valor estimado de sueño con 0 horas de internet.


R² ≈ 0.448. Esto indica que aproximadamente el 45% de la variabilidad en el sueño profundo puede explicarse por el tiempo que los niños pasan en internet.

c) Ahora suponga que solo se distingue entre pasar más de 1 hora en internet (X = 1)
y pasar una hora o menos en internet (X = 0). Estime nuevamente el modelo
lineal y compare los resultados. ¿Cómo puede interpretarse ahora ˆ β? Describa
cómo cambia ˆ β si se codifica con 0 a quienes pasan más de una hora en internet
y con 1 a los dem´as.

In [6]:
# Regresión binaria (X = 1 si >1h, 0 si <=1h)
X_bin = (internet > 1).astype(int).reshape(-1, 1)
reg_bin = LinearRegression().fit(X_bin, y)
y_pred_bin = reg_bin.predict(X_bin)
r2_bin = r2_score(y, y_pred_bin)

print("\nModelo binario:")
print(f"Beta_0 (intercepto): {reg_bin.intercept_:.3f}")
print(f"Beta_1 (diferencia promedio): {reg_bin.coef_[0]:.3f}")
print(f"R²: {r2_bin:.3f}")


Modelo binario:
Beta_0 (intercepto): 5.940
Beta_1 (diferencia promedio): -0.865
R²: 0.447


### Interpretación:
β₀: 5.94 → promedio de sueño en niños que usan ≤1h de internet.

β₁: -0.865 → los niños que usan >1h de internet duermen 0.865 horas menos que los demás.

R² ≈ 0.447 → el ajuste del modelo binario es prácticamente igual al continuo.